optimum/habana/transformers/generation/stopping_criteria.py

```diff
@@ -67,6 +68,7 @@ def gaudi_MaxTimeCriteria_call(
 def gaudi_EosTokenCriteria_call(
     self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs
 ) -> Union[torch.BoolTensor, bool]:
+    htcore.mark_step()
```

Comment:
this is a general change, and may affect performance in other models.

Can you add some numbers for the compile times u have with and without this change?

Reply:
is it possible to add teh markstep at teh end of the qwen model or something like that so the effect of this is isolated or qwen only?


otherwise we'd have to do a larger study to make sure it doesnt have any bad effect on any other model/usecase

Reply:
mark_step at the end of the model doesn't have the same effect. 
There is 2x improvement in warmup time. 
I agree this should be independent PR with more tests.
I will remove this particular change.